extensions/mongodb-client/runtime/src/main/java/io/quarkus/mongodb/runtime/MongoClients.java

```diff
@@ -384,6 +387,21 @@ private AuthenticationMechanism getAuthenticationMechanism(String authMechanism)
         return providers;
     }
 
+    private List<CommandListener> getCommandListeners(List<String> classNames) {
+        List<CommandListener> listeners = new ArrayList<>();
+        for (String name : classNames) {
+            try {
+                Class<?> clazz = Thread.currentThread().getContextClassLoader().loadClass(name);
```

Comment:
Could we use `Class.forName` here instead?

Reply:
Ok. Tested on a project with the jar and the native image does not make the difference, I suppose because we do
the initialization immediately after loading the class.